web/src/app/admin/AdminSMSSend.tsx

```diff
@@ -43,22 +55,43 @@ const useStyles = makeStyles({
 export default function AdminSMSSend(): JSX.Element {
   const classes = useStyles()
   const [cfgFromNumber] = useConfigValue('Twilio.FromNumber')
+  const [messageID, setMessageID] = useState('')
   const [fromNumber, setFromNumber] = useState(cfgFromNumber as string)
   const [toNumber, setToNumber] = useState('')
   const [body, setBody] = useState('')
   const [showErrorDialog, setShowErrorDialog] = useState(false)
 
-  const [send, sendStatus] = useMutation(sendSMSMutation, {
-    variables: {
-      input: {
-        from: fromNumber,
-        to: toNumber,
-        body,
+  const [send, { data: smsData, loading: smsLoading, error: smsError }] =
+    useMutation(sendSMSMutation, {
+      variables: {
+        input: {
+          from: fromNumber,
+          to: toNumber,
+          body,
+        },
       },
+      onError: () => setShowErrorDialog(true),
+      onCompleted: (data) => setMessageID(data.debugSendSMS.id),
+    })
+
+  const { data } = useQuery(debugMessageStatusQuery, {
+    variables: {
+      input: { providerMessageID: messageID },
     },
-    onError: () => setShowErrorDialog(true),
+    pollInterval: POLL_INTERVAL,
+    skip: !messageID,
   })
 
+  const getFromNumber = (): string => {
+    if (smsData?.debugSendSMS?.fromNumber) {
+      return smsData?.debugSendSMS?.fromNumber
+    }
+    if (data?.debugMessageStatus?.messageStatus?.formattedSrcValue) {
+      return data?.debugMessageStatus?.messageStatus?.formattedSrcValue
+    }
```

Comment:
Probably have the status be the source of truth -- could also make the argument to _only_ use the status number and ignore `fromNumber` from `debugSendSMS` -- then we get consistent formatting too.

```suggestion
    if (data?.debugMessageStatus?.messageStatus?.formattedSrcValue) {
      return data?.debugMessageStatus?.messageStatus?.formattedSrcValue
    }
    if (smsData?.debugSendSMS?.fromNumber) {
      return smsData?.debugSendSMS?.fromNumber
    }
```